src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -882,26 +1093,76 @@ def _get_dockerfile_content(repo, branch, token, source, context_path, dockerfil
             resource_group.name = parse_resource_id(matched_apps[0]["id"])[
                 "resource_group"
             ]
             env.set_name(matched_apps[0]["properties"]["environmentId"])
+            if env.is_connected_environment():
+                env.custom_location_id = matched_apps[0]["extendedLocation"]["name"]
         elif len(matched_apps) > 1:
             raise ValidationError(
                 f"There are multiple containerapps with name {name} on the subscription. "
                 "Please specify which resource group your Containerapp is in."
             )
 
 
+def _filter_containerapps_by_connected_cluster_id(cmd, matched_apps, connected_cluster_id):
+    result_apps = []
+    for app in matched_apps:
+        if app.get("extendedLocation"):
+            custom_location_from_app = get_custom_location(cmd=cmd, custom_location_id=app["extendedLocation"]["name"])
+            if custom_location_from_app and connected_cluster_id.lower() == custom_location_from_app.host_resource_id.lower():
+                result_apps.append(app)
+    return result_apps
+
+
+# If the connected cluster already have a custom location (with ext namespace) bind to the container app ext, uses it.
+def _prepare_custom_location_args(
+        cmd,
+        custom_location: "CustomLocation",
+        extension: "Extension"
+):
+    if custom_location.name is None:
+        if custom_location.connected_cluster_id is None:
+            raise ValidationError("please specify one of --connected-cluster-id or --custom-location you want to create the Connected Environment or specify the existing Connected Environment with --environment.")
+
+        connected_cluster = get_connected_k8s(cmd, custom_location.connected_cluster_id)
+        custom_location.location = connected_cluster.location
+        extension_list = list_cluster_extensions(cmd, connected_cluster_id=custom_location.connected_cluster_id)
+        for e in extension_list:
+            if e.extension_type.lower() == CONTAINER_APP_EXTENSION_TYPE:
+                extension.exists = True
+                extension.name = e.name
+                extension.namespace = e.scope.cluster.release_namespace
+                custom_location.cluster_extension_id = e.id
+                custom_location.namespace = e.scope.cluster.release_namespace
+                custom_location_list = list_custom_location(cmd,
+                                                            connected_cluster_id=custom_location.connected_cluster_id)
+                for c in custom_location_list:
+                    if e.id in c.cluster_extension_ids and e.scope.cluster.release_namespace == c.namespace:
+                        custom_location.set_name(c.id)
+                        custom_location.exists = True
+                        break
+                break
+    else:
+        custom_location.location = get_custom_location(cmd, custom_location_id=custom_location.get_rid()).location
```

Comment:
do not need to populate other information, e.g. ns, name?

Reply:
In the else statement, the custom_location.name is not None. It also mean the --custom-location is part of input arguments.

Reply:
but what if the customer location doesn't exist?